datastruct/elasticarray.h

```diff
@@ -159,6 +166,13 @@ int elasticarray_exportdup(struct elasticarray *, void **, size_t *, size_t);
 		return (rec);						\
 	}								\
 	static inline void						\
+	prefix##_iter(struct prefix##_struct * EA,			\
+	    void(free_fp)(void *))					\
```

Comment:
This should be a `void(free_fp)(rectype *)` function pointer, with appropriate casting so that `elasticarray_iter` gets the `void (*)(void *)` function pointer type it expects.